euphonic/powder.py

```diff
@@ -0,0 +1,279 @@
+"""Functions for averaging spectra in spherical q bins"""
+
+import numpy as np
+from pint import Quantity
+from typing import Union
+
+from euphonic import (Crystal, DebyeWaller, ForceConstants, QpointPhononModes,
+                      Spectrum1D)
+from euphonic import ureg
+from euphonic.util import mp_grid
+
+sampling_choices = {'golden', 'sphere-projected-grid', 'spherical-polar-grid',
+                    'spherical-polar-improved', 'random-sphere'}
+
+
+def sample_sphere_dos(fc: ForceConstants,
```

Comment:
What's the use case for this function? Would this form part of creating a 2D (|Q|, E) dos spectrum by calling this multiple times? Is that something people would want to do?

Reply:
Yes, this could be called repeatedly to generate a 2D DOS.

Is there any value in mod(q)-constrained/resolved DOS data? Well, I guess it's more of a general theoretical tool than something for simulating experiments. If you are interested in the general properties of phonon dispersion it's nice to separate this from the neutron-weighted structure factor data. We'll probably use it as part of the work on convergence.